packages/opentelemetry-resource-util/src/index.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Resource} from '@opentelemetry/resources';
+import {ResourceAttributes} from '@opentelemetry/semantic-conventions';
+
+const GCP_GCE_INSTANCE = 'gce_instance';
+const AWS_EC2_INSTANCE = 'aws_ec2_instance';
+
+export interface MonitoredResource {
+  type: string;
+  labels: {[key: string]: string};
+}
+
+/**
+ * Given an OTel resource, return a MonitoredResource. If any field is missing,
+ * return the global resource. The only currently supported monitored resources
+ * are gce_instance and aws_ec2_instance.
+ * @param resource
+ * @param projectId
+ */
+export function mapOtelResourceToMonitoredResource(
+  resource: Resource,
+  projectId: string
+): MonitoredResource {
+  const cloudProvider = `${
```

Comment:
Instead of string-interpolating this, could we say

```ts
const cloudProvider = resource.attributes[ResourceAttributes.CLOUD_PROVIDER] as string | undefined;
```

Otherwise the undefined case gets mapped to the string `"undefined"` which is more likely to trip us up later.

Reply:
good catch